alchemist-engine/src/main/java/it/unibo/alchemist/core/implementations/Engine.java

```diff
@@ -590,29 +595,34 @@ private NodeAddition(final Node<T> source) {
 
     private abstract class UpdateOnReaction extends Update {
 
-        private final Reaction<T> actualSource;
+        private final Actionable<T> actualSource;
 
         private UpdateOnReaction(final Reaction<T> source) {
             super(source.getNode());
             actualSource = source;
         }
 
+        private UpdateOnReaction(final Actionable<T> source) {
+            super(null);
```

Comment:
This is a call for NPEs, plus constructor calls are linked at compile-time... We need to revise the `Update` interface